scripts/checkDeps.ps1

```diff
@@ -65,9 +71,22 @@ if ($_packagesToInstall.Count -gt 0) {
         # make sure to update the list first
         sudo apt-get update
 
+        # ok docker and docker-compose exist so let's check the packages
+        $_installedScrips = New-Object Collections.Generic.List[string]
+
+        # If it is a script, execute the scrip, else apt-get install the package
         foreach ($item in $_packagesToInstall) {
-            sudo apt-get install -y $item
-            
+            if ($item.EndsWith('.sh')) {
```

Comment:
I would not like to force the template developer to use only shell scripts. So, my proposal for this feature would be add a new property on the `deps.json` the `installScripts`, should look like:

```json
{
    "packages": [
        "openssh-client",
        "sshpass",
        "build-essential",
        "make",
        "gdb",
        "gcc"
    ],
    "installScripts": [
        "./install-mono.sh",
        "/home/user/libs/install-lib.ps1",
        "/root/binaryname"
    ]
}
```

This new property should be an array that accepts absolute or relative to `.conf` paths.
Also, would be good to store the specific install scripts on the `.conf` folder of the specific template.

Let me know what you think.

Reply:
About the scrips, I had added just powershell and bash, but I agree it is better to have them as a separated property and support any type of script, as long as it is executed in the normal `./$scriptName` way of executing scripts

Reply:
About them being stored in each template `.conf` folder, it has the good side of being easier to understand where the script is in this templates repo. But it has the downside of maintenance, the `install-dotnet-sdk-8.sh` for example is used on 6 templates, meaning that if we change anything in this script we will need to remember to change it is all of them.

Reply:
hmm, yeah, you have a point. What about using variables then? so, templates developers should be able to do this:

```
{
    "packages": [
        "openssh-client",
        "sshpass",
        "build-essential",
        "make",
        "gdb",
        "gcc"
    ],
    "installScripts": [
        "./install-mono.sh",
        "${HOME}/.apollox/scripts/install-lib.ps1",
        "/root/binaryname"
    ]
}
```

The main idea here is to be able to detach the workspace from the `.apollox` folder. I think that using variables we can make easy if in some case the user want to detach from the `.apollox` folder by copying the script and changing the `deps.json`.

Reply:
to start let's support only `${HOME}`, and if needed we add more ...

Reply:
I updated here the commit. I made it in a way that it supports both scripts present just in the template and also common scripts, which are copied from the `scripts/install-deps-scripts` to the `.conf/install-deps-scripts` of the project when the project is created. Also, I added it in a way that of for example the template already have this `.conf/install-deps-scripts/$script` modified to his specific use case (let's say for example that Avalonia uses a modified `install-dotnet-sdk-6.sh` for example) it gives priority to the one already present in the template.